valhalla/baldr/double_bucket_queue.h

```diff
@@ -82,16 +112,25 @@ class DoubleBucketQueue final {
   void clear() {
     // Empty the overflow bucket and each bucket
     overflowbucket_.clear();
-    while (currentbucket_ != buckets_.end()) {
-      currentbucket_->clear();
-      currentbucket_++;
+    for (auto& b : buckets_) {
+      b.clear();
```

Comment:
i think we need to also need to reduce the size of the buckets and the overflowbucket if it goes over some configurable value otherwise they will grow infinitely

Reply:
I actually do the same thing in reuse() method. I'm not yet sure about the purpose of clear(), it seems it was never called, except tests. To free all allocated memory it's better to call `buckets_.clear()`, but in this case memory cannot be reused.